test/ha/ha.go

```diff
@@ -102,3 +104,18 @@ func readyEndpointsDoNotContain(ip string) func(*corev1.Endpoints) (bool, error)
 		return true, nil
 	}
 }
+
+func isInternalEncryptionOn(t *testing.T, client kubernetes.Interface) bool {
+	var cm *corev1.ConfigMap
+	var err error
+	if cm, err = client.CoreV1().ConfigMaps(system.Namespace()).Get(context.Background(), "config-network", metav1.GetOptions{}); err != nil {
+		t.Fatalf("Failed to get cm config-network: %v", err)
+	}
+	netCfg, err := netcfg.NewConfigFromMap(cm.Data)
+	if err != nil {
+		t.Fatalf("Failed to construct network config: %v", err)
+	}
+
+	return netCfg.ExternalDomainTLS || netCfg.SystemInternalTLSEnabled() || (netCfg.ClusterLocalDomainTLS == netcfg.EncryptionEnabled) ||
```

Comment:
This condition is duplicate with what main has, we can improve in a followup PR.